App.js

```diff
@@ -231,51 +249,94 @@ const navTheme = {
   },
 };
 
+const IsSmallScreenContext = React.createContext();
+
+const createMainStack = (Stack) => () => {
+  return (
+    <Stack.Navigator
+      initialRouteName="LeftHandNav"
+    >
+        <Stack.Screen
+          name="LeftHandNav"
+          component={LeftHandNav}
+          options={{headerShown: false}}
+        />
+        <Stack.Screen
+          name="Chat"
+          component={ChatScreen}
+          options={{headerShown: false}}
+          initialParams={{ id: 1 }} />
+        <Stack.Group>
+          <Stack.Screen
+            name="Search"
+            component={SearchScreen}
+            options={{headerShown: false}}
+          />
+          <Stack.Screen
+            name="SettingsStack"
+            component={SettingsStackNavigator}
+            options={{headerShown: false}}
+          />
+        </Stack.Group>
+    </Stack.Navigator>
+  )
+}
+
+const WebStack = createWebNavigator();
+const NativeStack = createNativeStackNavigator();
+
+const SettingsStack = createNativeStackNavigator();
+
+const WebNavigator = createMainStack(WebStack);
+const NativeNavigator= createMainStack(NativeStack);
+
+const RootStack = createNativeStackNavigator();
```

Comment:
Hi, @adamgrzybowski. Can you share your thought process on why we are adding an extra layer of a navigator? I am having a bit of a hard time understanding this change, so if you can give some context that will be cool.

Reply:
In high-level it seems like this allows us to not use the patch (its gone now). But we definitely still need to get down to the keys problem (and why would be solved with the wrapping navigator)

Reply:
Sure! 
1. The conditionally rendered navigators aren't really a pattern in `react-navigation` unless they are wrapped in a screen. With this approach, we are closer to the solutions presented in the react-navigation docs. That's also something my colleague pointed me. Being closer to the general use case we are less prone to bugs.
2. If we are conditionally rendering screens and not navigators that means we keep the same root navigator. The key for the root stays the same after resizing and we don't need to patch `react-navigation` to make it work when navigating back with the browser button. 
3. It will be easier to separate the logic for resizing. I haven't done it here yet but I guess that may be useful in the main app. Instead of `RootNavigator` we could call it `ResponsiveNavigator` or something like that.

Reply:
Looks promising, waiting for point 3 to be implemented. Going to give it some more testing for now.